### ISSUE TITLE
Warning: Cannot read property 'indexOf' of undefined Used --force, continuing.

### ISSUE THREAD

**Issue by pawankorotane** [balderdashy/sails#2915]
HI Guys,

I am getting these warnings when I runs the sails lift command it works fine in my development machine (mac) when I run my application on AWS EC2(ubuntu)  I am getting indexOf warnings in could of task like jst, sails-linker. Could you please help me out to fix this problem.
here is my package.json file and I am using Sails V.0.10

{
  "dependencies": {
    "bcrypt": "~0.8.0",
    "connect-redis": "^1.4.5",
    "ejs": "~0.8.4",
    "grunt": "0.4.2",
    "grunt-bower-task": "^0.3.4",
    "grunt-contrib-clean": "~0.5.0",
    "grunt-contrib-coffee": "~0.10.1",
    "grunt-contrib-concat": "~0.3.0",
    "grunt-contrib-copy": "~0.5.0",
    "grunt-contrib-cssmin": "~0.9.0",
    "grunt-contrib-jst": "~0.6.0",
    "grunt-contrib-less": "0.11.1",
    "grunt-contrib-uglify": "~0.4.0",
    "grunt-contrib-watch": "~0.5.3",
    "grunt-sails-linker": "~0.9.5",
    "grunt-sync": "~0.0.4",
    "include-all": "~0.1.3",
    "jwt-simple": "~0.2.0",
    "nodemailer": "~1.3.0",
    "nodemailer-smtp-transport": "~0.1.13",
    "rc": "~0.5.0",
    "request": "~2.51.0",
    "sails": "~0.10.5",
    "sails-disk": "~0.10.0",
    "sails-generate-bower": "0.0.2",
    "sails-mongo": "~0.10.5",
    "sails-redis": "^0.10.3",
    "sails-validation-messages": "0.0.0"
  },
  "name": "dating_room",
  "private": true,
  "version": "0.0.0",
  "description": "a Sails application",
  "keywords": [],
  "scripts": {
    "start": "sails lift",
    "debug": "node debug app.js"
  },
  "main": "app.js",
  "repository": {
    "type": "git",
    "url": "**********t"
  },
  "author": "*****",
  "license": ""
}

Running "bower:install" (bower) task
>> Installed bower packages
>> Copied packages to /home/ubuntu/node_projects/dating_room/assets/vendor

Running "clean:dev" (clean) task

Running "jst:dev" (jst) task
Warning: Cannot read property 'indexOf' of undefined Used --force, continuing.

Running "less:dev" (less) task
File .tmp/public/styles/importer.css created: 0 B → 79.95 kB

Running "copy:dev" (copy) task
Copied 172 files

Running "coffee:dev" (coffee) task

Running "sails-linker:devJs" (sails-linker) task
Warning: Cannot read property 'indexOf' of undefined Used --force, continuing.

Running "sails-linker:devStyles" (sails-linker) task
Warning: Cannot read property 'indexOf' of undefined Used --force, continuing.

Running "sails-linker:devTpl" (sails-linker) task
padding length 4
File "views/layout.ejs" updated.

Running "sails-linker:devJsJade" (sails-linker) task
Warning: Cannot read property 'indexOf' of undefined Used --force, continuing.

Running "sails-linker:devStylesJade" (sails-linker) task
Warning: Cannot read property 'indexOf' of undefined Used --force, continuing.

Running "sails-linker:devTplJade" (sails-linker) task

**Comment by aato** [comment]
It's a wild guess but check your file permissions for your .tmp folder, especially the .tmp/public/jst.js file (since that's what jst:dev is trying to require when you get the error).

**Comment by pawankorotane** [comment]
Hi aato,

thanks for the reply I have set the permission to 0777 still not luck.

**Comment by pawankorotane** [comment]
Now getting this error

Verifying property jst.dev exists in config...OK
Warning: Cannot read property 'indexOf' of undefined Use --force to continue.

**Comment by tjwebb** [comment]
What version of node are you running, and which 0.10 release of sails? There is 0.10.1, 0.10.2, etc...

**Comment by pawankorotane** [comment]
I am using node version 0.12.2 and sails 0.10.5 thanks

**Comment by tjwebb** [comment]
Node 0.12 didn't exist when sails 0.10.5 was released. Can you try
installing the latest version of sails (0.11)?

**Comment by sailsbot** [comment]
Thanks for posting, @pawankorotane.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
